src/Storages/MergeTree/MergeTreeIndexText.h

```diff
@@ -297,9 +304,30 @@ struct MergeTreeIndexTextGranuleBuilder
     std::unique_ptr<Arena> arena;
 };
 
+class MergeTreePreprocessor
+{
+public:
+    MergeTreePreprocessor(const String & expression, const IndexDescription & index_description);
+    std::pair<ColumnPtr,size_t> processColumn(const ColumnWithTypeAndName & index_column_with_type_and_name, size_t start_row, size_t n_rows) const;
+    String processString(const String &input) const;
```

Comment:
Needs a source code comment, "processString" is too generic to understand from the name what it is doing.